scala-package/packageTest/README.md

```diff
@@ -0,0 +1,72 @@
+
+You can also run `make scalainstall` to install from a local build.
+
+After installing, run `make testinstall` in the package test directory to run the tests.  Note that unless you also install an additional mxnetexamples jar, you can only run the unit tests.
+
+### Test Local Deployment
+
+To test the jars that would be produced by a deployment, you can run `make scaladeploylocal` from the main mxnet directory. This produces a local snapshot located at `scala-package/local-snapshot`. To test this local snapshot, run `make testlocal`.
+
+### Remote Repository Snapshot
+
+This mode is to test a jar located in a remote repository. The default repository is the apache snapshot repisotory located at `https://repository.apache.org/content/repositories/snapshots`. Note that the actual jar in a repisotory should be located at `$repoUrl/org/apache/mxnet/mxnet-full_$scalaVersion-$osMode/$version/*.jar`.
+
+Test the snapshot repo using `make testsnapshot` or a different repo using `make testsnapshot MXNET_REPO=$NEW_REPO_URL`.
+
+### Options
+
+You are able to run unit tests, integration tests, or both using this utility. To run the unit tests, add the flag `UNIT=1` to make (e.g. `make testsnapshot UNIT=1`). Use `INTEGRATION=1` for integration tests. The default behavior is to run both the unit and integration tests. However, the integration tests require that the mxnet examples be installed in addition to the full mxnet package (see test mode instructions above).
+
+An additional option, you can specify the mxnet version with `MXNET_VERSION=1.3.1-SNAPSHOT`.
+
+## Cleaning Up
+
+You can clean temporary files and target artifacts by running `make scalaclean`.
+
+## Troubleshooting
+
+### Missing Examples
+
+If you fail with the following error
+```
+[ERROR] Failed to execute goal org.scalatest:scalatest-maven-plugin:1.0:test (test) on project mxnet-scala-packagetest-examples_2.11: There are test failures -> [Help 1]
+[ERROR]
+[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
+[ERROR] Re-run Maven using the -X switch to enable full debug logging.
+[ERROR]
+[ERROR] For more information about the errors and possible solutions, please read the following articles:
+[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
+[ERROR]
+[ERROR] After correcting the problems, you can resume the build with the command
+[ERROR]   mvn <goals> -rf :mxnet-scala-packagetest-examples_2.11
+Makefile:57: recipe for target 'scalaintegrationtest' failed
+make: *** [scalaintegrationtest] Error 1
+```
+
+and stacktrace begins with the following,
+
+```
+*** RUN ABORTED ***
+  java.lang.NoClassDefFoundError: org/apache/mxnetexamples/Util$
+```
+
+you are missing the mxnetexamples package.  See your test mode installation section for details.
```

Comment:
Can you link to the test mode installation section so that it is easy to navigate ?

Reply:
In document linking is not officially part of markdown. Github added the feature, but it won't work in other viewers such as intellij. I would rather have no link than a link that might not work

Reply:
Okay! Makes sense